aiohttp/web_exceptions.py

```diff
@@ -420,18 +420,21 @@ class HTTPUnavailableForLegalReasons(HTTPClientError):
 
     def __init__(
         self,
-        link: StrOrURL,
+        link: Optional[StrOrURL] = None,
         *,
         headers: Optional[LooseHeaders] = None,
         reason: Optional[str] = None,
         text: Optional[str] = None,
         content_type: Optional[str] = None,
     ) -> None:
+        if link is not None and not link:
+            raise ValueError("link argument cannot be empty.")
         super().__init__(
             headers=headers, reason=reason, text=text, content_type=content_type
         )
-        self.headers["Link"] = f'<{str(link)}>; rel="blocked-by"'
-        self._link = URL(link)
+        self._link = URL(link) if link else None
+        if self.link:
+            self.headers["Link"] = f'<{str(self.link)}>; rel="blocked-by"'
```

Comment:
```suggestion
        self._link = None
        if self.link:
            self._link = URL(link)
            self.headers["Link"] = f'<{str(self.link)}>; rel="blocked-by"'
```
Maybe avoid the duplicate check?

Reply:
This depends on the previous comment's outcome, but can do of course. 👍